CMSgov/easi-app
# [EASI-2885] add state and step fields to intakes

# EASI-2885

## Changes and Description

- Adds a `state` field to System Intakes that represents whether an intake is open or closed.
- Adds a `step` field to System Intakes that indicates which part of the process the intake is currently in.

## How to test this change

Start the backend with `scripts/dev up:backend`, then open the [GQL Playground](http://localhost:8080/api/graph/playground#) and run the following mutation to create a System Intake:

```gql
mutation($input: CreateSystemIntakeInput!) {
  createSystemIntake(input: $input) {
    id
    requester {
      name
    }
    step
    state
    status
  }
}
```

with this variable data:

```json
{
  "input": {
    "requestType": "NEW",
    "requester": {
      "name": "Bob"
    }
  }
}
```

You should see the new returned `state` and `step` fields.

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
